### ISSUE TITLE
Great library, any way to get the progress event?

### ISSUE THREAD

**Issue by mayeaux** [jamesloper/easy-backblaze#1]
I'd like to feed back progress to the frontend, do you plan on offering support for that API?

**Comment by lemontron** [comment]
Yes I'll be sure to add that right after fs input stream support :)

**Comment by mayeaux** [comment]
I was just going to say that that's the other part that's needed ;P Okay cool I'll keep an eye on the lib thanks!

**Comment by mayeaux** [comment]
How would you go about adding support for a stream? I could really use that in the code I'm writing right now. Thanks!

**Comment by lemontron** [comment]
Almost done with both improvements. A sneak peak :)

## Get Progress Updates

``` javascript
var client = b2.uploadFile('/var/tmp/test.mp4', {
    name: 'swiggity-swooty.mp4' // Optional, can override the file name
    bucket: 'swooty' // Optional, a bucketName, default is the first bucket
}, function(err, res) {
    console.log('Done!', err, res);
});

client.on('progress', function(progress) {
    console.log('Progress:', progress);
});
```

**Comment by mayeaux** [comment]
Awesome! Looking forward to it man, much appreciated!

**Comment by lemontron** [comment]
Done -> https://www.npmjs.com/package/easy-backblaze

Big upgrade also - removed dependency on `request`!

However...
Back when I initially made this, I tested it by shooting off 100's of uploads. Some of them would always send HTTP errors and that's how I tested the auto retry feature. It seems I can't overload b2 right now, it's being a lot more stable these days. 

So basically, I released it anyway, but stay tuned for bug fixes, not sure how reliable it will be if the auto retry doesn't work as well as it used to.
